gui/include/euRun.hh

```diff
@@ -31,126 +31,142 @@ inline std::string to_bytes(const std::string & val) {
-      m_prevtrigs = 0;
-      m_prevtime = 0.0;
-      m_runstarttime = 0.0;
-      StartRun(cont ? "Continued" : txtRunmsg->displayText().toStdString());
-      EmitStatus("RUN", to_string(m_runnumber));
-      emit StatusChanged("EVENT", "0");
-      emit StatusChanged("TRIG", "0");
-      emit StatusChanged("PARTICLES", "0");
-      emit StatusChanged("RATE", "");
-      emit StatusChanged("MEANRATE", "");
-      SetState(ST_RUNNING);
-    }
-    void on_btnStop_clicked() {
-      StopRun();
-      EmitStatus("RUN", "(" + to_string(m_runnumber) + ")");
-      SetState(ST_READY);
-    }
-    void on_btnLog_clicked() {
-      std::string msg = txtLogmsg->displayText().toStdString();
-      EUDAQ_USER(msg);
     }
-    void timer() {
-      if (!m_stopping) {
-        if (m_runsizelimit >= 1024 && m_filebytes >= m_runsizelimit) {
-          EUDAQ_INFO("File limit reached: " + to_string(m_filebytes) + " > " + to_string(m_runsizelimit));
-          eudaq::mSleep(1000);
-          StopRun(false);
-          eudaq::mSleep(8000);
-          while (m_producerbusy) {
-            eudaq::mSleep(50);
-          }
-          on_btnStart_clicked(true);
-        } else if (dostatus) {
-          GetStatus();
-        }
-      }
+    if (m_startrunwhenready && !m_producer_pALPIDEfs_not_ok && !m_producer_pALPIDEss_not_ok) {
+      m_startrunwhenready = false;
+      on_btnStart_clicked(false);
     }
-    void ChangeStatus(const QString & name, const QString & value) {
-      status_t::iterator i = m_status.find(name.toStdString());
-      if (i != m_status.end()) {
-        i->second->setText(value);
-      }
+  }
+  void ChangeStatus(const QString & name, const QString & value) {
+    status_t::iterator i = m_status.find(name.toStdString());
+    if (i != m_status.end()) {
+      i->second->setText(value);
     }
-	void btnLogSetStatusSlot(bool status){
+  }
+  void btnLogSetStatusSlot(bool status){
 
-		 btnLog->setEnabled(status);
-	}
+    btnLog->setEnabled(status);
+  }
 	
 signals:
-    void StatusChanged(const QString &, const QString &);
-	void btnLogSetStatus(bool status);
-	void SetState(int status);
-  private:
-    RunControlModel m_run;
-    RunConnectionDelegate m_delegate;
-    QTimer m_statustimer;
-    typedef std::map<std::string, QLabel *> status_t;
-    status_t m_status;
-    int m_prevtrigs;
-    double m_prevtime, m_runstarttime;
-    int64_t m_filebytes;
-    bool dostatus;
+  void StatusChanged(const QString &, const QString &);
+  void btnLogSetStatus(bool status);
+  void SetState(int status);
+private:
+  RunControlModel m_run;
+  RunConnectionDelegate m_delegate;
+  QTimer m_statustimer;
+  typedef std::map<std::string, QLabel *> status_t;
+  status_t m_status;
+  int m_prevtrigs;
+  double m_prevtime, m_runstarttime;
+  long long m_filebytes;
+  long long m_events;
+  bool dostatus;
+  bool m_producer_pALPIDEfs_not_ok;
+  bool m_producer_pALPIDEss_not_ok;
+  bool m_startrunwhenready;
```

Comment:
Thanks for the diff, a few comments on that:

```
@@ -74,13 +73,9 @@ class RunControlGUI : public QMainWindow, public Ui::wndRun, public eudaq::RunCo
 
   void SetStateSlot(int state) {
     btnConfig->setEnabled(state != ST_RUNNING);
-			btnTerminate->setEnabled(state != ST_RUNNING);
     btnStart->setEnabled(state == ST_READY);
     btnStop->setEnabled(state == ST_RUNNING);
   }
-
-  void on_btnTerminate_clicked() { close(); }
-
   void on_btnConfig_clicked() {
```
Does this solve a problem you experienced? What is the reason behind these changes?

```
@@ -151,6 +163,10 @@ signals:
   status_t m_status;
   int m_prevtrigs;
   double m_prevtime, m_runstarttime;
-    int64_t m_filebytes;
+  long long m_filebytes;
+  long long m_events;
```
We try to stick to fixed size integers to keep maximum compatibility. Is there a specific reason for changing to `long long` or would `uint64_t` also be fine with you?

```
+  bool m_producer_pALPIDEfs_not_ok;
+  bool m_producer_pALPIDEss_not_ok;
[...]
+    else if (id.GetName() == "pALPIDEfs") {
+      m_producer_pALPIDEfs_not_ok = (status->GetLevel() != 1);
+    }
+    else if (id.GetName() == "pALPIDEss") {
+      m_producer_pALPIDEss_not_ok = (status->GetLevel() != 1);
+    }
+  }
```
Is this really necessary? I know that the current finite state machine of the RunControl is somehow broken, but we will have a Summer Student working on this. I hope that after that we will always be in a well-defined state and the producers returning error messages will not just be a nice GUI plotting feature.